### PR TITLE
Remove old version code

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

**Review by awemany — APPROVED** [review]
Nice! Looks good to me, though didn't compile or test.

File: src/main.cpp [hunk 1]
```diff
@@ -1028,11 +1028,8 @@ static void handleAddressAfterInit(CNode *pfrom)
         }
 
         // Get recent addresses
-        if (pfrom->fOneShot || pfrom->nVersion >= CADDR_TIME_VERSION || addrman.size() < 1000)
-        {
```
Comment (sickpig) [inline src/main.cpp]: Shoudn't we still need to check for `pfrom->fOneShot` or `addrman.size() < 1000` ?

Comment (Greg-Griffith) [reply src/main.cpp]: since pfrom->nVersion was always higher than CADDR_TIME_VERSION this always evaluated to true so i just removed it.

Comment (sickpig) [reply src/main.cpp]: you are right indeed

**Review by sickpig — APPROVED** [review]

**Comment by ptschip** [comment]
how do you know pfrom->nVersion is always higher...you can't know which
client will connect to you or what nVersion they will have?

**Comment by Greg-Griffith** [comment]
@ptschip CADDR_TIME_VERSION was added in 2012 by to bitcoin version 0.6 ish. bip00031 is also from around the same time, i don't think any clients exist anymore from before that version. especially those that also comply with the rules of our HFs since BCH forked from BTC. Even if someone somehow doesnt have the GETADDR message that we are responding with, it should just fall through as an unrecognized message anyway

**Comment by sickpig** [comment]
> how do you know pfrom->nVersion is always higher...you can't know which client will connect to you or what nVersion they will have?

@ptschip you have a point.

Having said that, all the clients that implemented BCH so far have forked Core at point where `nVersion` was higher than `CADDR_TIME_VERSION`.

For other independent implementations I've checked libbitcoin/bitprim and BCHD and both have a version higher than `CADDR_TIME_VERSION`.

As a last point I have checked the data gathered by BU seeder and this what I found:

``` 
# nodes nVersion 
      4 70013
      2 70014
   1958 70015
     17 80002
    942 80003
```

**Comment by sickpig** [comment]
As #1487, this PR had `checkdatasig` and `invalidrequest` python tests that are failing with the same assertion: "Not all nodes requested block" (see https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/458243364)

So it seems that the failure is due to some changes unrelated this PR. 

edit: I'm not able to reproduce the problem locally on current `dev`

**Comment by sickpig** [comment]
@Greg-Griffith I dunno why but this PR continue to fails the python tests when run in Travis no matter how many times I try to restart jobs. This is the error message:

```
2018-11-28 09:29:52,597.ERROR: Assertion failed: Not all nodes requested block```

In last Travis session the failings tests are:

- `invalidtxrequest.py`
- `checkdatasig_activation.py` 

I thought that @gandrewstone fixed the problem in #1487

Trying to reproduce locally.

**Comment by sickpig** [comment]
@Greg-Griffith 

I'm able to reproduce the error locally, e.g. `checkdatasig_activation.py` fails in reproducible way: 

```
~/src/bu-sickpig(remove-old-version-markers)$> qa/rpc-tests/checkdatasig_activation.py 
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:13936
11/28/2018 02:56:07 PM ERROR: Assertion failed: Not all nodes requested block
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/test_framework.py", line 222, in main
    self.run_test()
  File "qa/rpc-tests/checkdatasig_activation.py", line 115, in run_test
    self.test.run()
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/comptool.py", line 370, in run
    self.sync_blocks(block.sha256, 1)
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/comptool.py", line 211, in sync_blocks
    raise AssertionError("Not all nodes requested block")
11/28/2018 02:56:07 PM ERROR: Failed
```

**Comment by sickpig** [comment]
same happens for `invalidtxrequest.py` : 

```
~/src/bu-sickpig(remove-old-version-markers)$> qa/rpc-tests/invalidtxrequest.py 
2018-11-28 14:58:31,511.INFO: Random seed: 1543413511
2018-11-28 14:58:31,511.INFO: Initializing test directory /tmp/test_invalidtxrequestpy_18ibj5i1/5437
2018-11-28 14:58:32,526.INFO: Started bitcoind 0 as pid 5438 at 127.0.0.1:14568 dir /tmp/test_invalidtxrequestpy_18ibj5i1/5437/node0  
MiniNode: Connecting to Bitcoin Node IP # 127.0.0.1:14568
2018-11-28 14:58:43,580.ERROR: Assertion failed: Not all nodes requested block
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/test_framework.py", line 222, in main
    self.run_test()
  File "qa/rpc-tests/invalidtxrequest.py", line 31, in run_test
    test.run()
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/comptool.py", line 370, in run
    self.sync_blocks(block.sha256, 1)
  File "/home/sickpig/src/bu-sickpig/qa/rpc-tests/test_framework/comptool.py", line 211, in sync_blocks
    raise AssertionError("Not all nodes requested block")
2018-11-28 14:58:43,582.INFO: Stopping nodes
2018-11-28 14:58:43,758.INFO: mininode network processing thread completed
2018-11-28 14:58:43,799.INFO: Not cleaning up dir /tmp/test_invalidtxrequestpy_18ibj5i1/5437
2018-11-28 14:58:43,799.ERROR: Failed
```

**Comment by sickpig** [comment]
@Greg-Griffith 

Ok I bisected the PR and problem has been introduce by this commit:

```
commit e50f1b02c32ca191602f039634bf1a9bc7f6be94
Author: Greg Griffith <griffith@cryptounited.io>
Date:   Tue Nov 20 14:53:23 2018 -0800

    remove code for CADDR_TIME_VERSION. all nodes are higher than this version
```

I guess this could be due to the fact that probably mininode is using a pre `CADDR_TIME_VERSION` protocol spec.

edit: seems not to be the case

```
nodemessages.py#L11  MY_VERSION = 60001  # past bip-31 for ping/pong
```

File: src/protocol.h [hunk 2]
```diff
@@ -386,7 +386,7 @@ class CAddress : public CService
         int nVersion = s.GetVersion();
         if (s.GetType() & SER_DISK)
             READWRITE(nVersion);
-        if ((s.GetType() & SER_DISK) || !(s.GetType() & SER_GETHASH))
+        if ((s.GetType() & SER_DISK) || (nVersion >= CADDR_TIME_VERSION && !(s.GetType() & SER_GETHASH)))
```
Comment (sickpig) [inline src/protocol.h]: I guess that now we need to remove `CADDR_TIME_VERSION` from the "unused network protocol versions" section and put it back where it was.

**Review by sickpig — CHANGES_REQUESTED** [review]
After the last commit I think that the PR is good to go, before merging we need to mark `CADDR_TIME_VERSION` as still used, see:
https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1486#discussion_r237383309

**Review by sickpig — APPROVED** [review]

File: src/version.h [hunk 3]
```diff
@@ -15,25 +15,12 @@ static const int PROTOCOL_VERSION = 80003;
 //! initial proto version, to be increased after version/verack negotiation
 static const int INIT_PROTO_VERSION = 209;
 
-//! In this version, 'getheaders' was introduced.
-static const int GETHEADERS_VERSION = 31800;
-
-//! disconnect from peers older than this proto version
-static const int MIN_PEER_PROTO_VERSION = GETHEADERS_VERSION;
```
Comment (gandrewstone) [inline src/version.h]: I think we still want this disconnect logic, for future changes...

File: src/version.h [hunk 4]
```diff
@@ -15,25 +15,12 @@ static const int PROTOCOL_VERSION = 80003;
...
-static const int NOBLKS_VERSION_START = 32000;
-static const int NOBLKS_VERSION_END = 32400;
-
-//! BIP 0031, pong message, is enabled for all versions AFTER this one
-static const int BIP0031_VERSION = 60000;
-
-//! "mempool" command, enhanced "getdata" behavior starts with this version
-static const int MEMPOOL_GD_VERSION = 60002;
+//! disconnect from peers older than this proto version
+static const int MIN_PEER_PROTO_VERSION = 31800;
```
Comment (gandrewstone) [inline src/version.h:23]: propose we bump this to NO_BLOOM_VERSION

**Review by gandrewstone — DISMISSED** [review]

**Pull request merged**